src/pd/retry.rs

```diff
@@ -109,20 +109,28 @@ impl fmt::Debug for RetryClient {
 }
 
 // A node-like thing that can be connected to.
+#[async_trait]
 trait Reconnect {
     type Cl;
-    fn reconnect(&self, interval: u64) -> Result<()>;
+    async fn reconnect(&self, interval: u64) -> Result<()>;
     fn with_cluster<T, F: Fn(&Self::Cl) -> T>(&self, f: F) -> T;
 }
 
+#[async_trait]
 impl Reconnect for RetryClient<Cluster> {
     type Cl = Cluster;
 
-    fn reconnect(&self, interval: u64) -> Result<()> {
-        if let Some(cluster) =
+    async fn reconnect(&self, interval: u64) -> Result<()> {
+        if let Some(cluster) = {
```

Comment:
I realise that it could make some mess if two `reconnect` happens at the same time. I think it's better to add a gate to make sure that if there is an on-going reconnecting progress, we don't trigger another one.